lib/pages/settings/blocks.dart

```diff
@@ -0,0 +1,279 @@
+                  child: Center(child: Text('error UwU')),
+                )
+              else ...[
+                for (final user in store.blockedUsers)
+                  _BlockPersonTile(user, key: ValueKey(user)),
+                if (store.blockedUsers.isEmpty)
+                  const ListTile(
+                      title: Center(child: Text('No users blocked'))),
+                // TODO: add user search & block
+                // ListTile(
+                //   leading: const Padding(
+                //     padding: EdgeInsets.only(left: 16, right: 10),
+                //     child: Icon(Icons.add),
+                //   ),
+                //   onTap: () {},
+                //   title: const Text('Block User'),
+                // ),
+                const Divider(),
+                for (final community in store.blockedCommunities)
+                  _BlockCommunityTile(community, key: ValueKey(community)),
+                if (store.blockedCommunities.isEmpty)
+                  const ListTile(
+                    title: Center(
+                      child: Text('No communities blocked'),
+                    ),
+                  ),
+                // TODO: add community search & block
+                // const ListTile(
+                //   leading: Padding(
+                //     padding: EdgeInsets.only(left: 16, right: 10),
+                //     child: Icon(Icons.add),
+                //   ),
+                //   onTap: () {},
+                //   title: Text('Block Community'),
+                // ),
+              ]
+            ],
+          ),
+        );
+      },
+    );
+  }
+}
+
+class _BlockPersonTile extends HookWidget {
+  final PersonSafe person;
+  const _BlockPersonTile(this.person, {Key? key}) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
+    final token = useAccountsStoreSelect(
+        (store) => store.defaultUserDataFor(person.instanceHost)!.jwt);
+
+    void showSnackBar(String e) {
+      ScaffoldMessenger.of(context).hideCurrentSnackBar();
+      ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(e)));
+    }
+
+    final pending = useState(false);
+
+    unblock() async {
+      if (pending.value) return;
+      pending.value = true;
+      try {
+        await LemmyApiV3(person.instanceHost).run(
+            BlockPerson(auth: token.raw, block: false, personId: person.id));
+        context.read<BlocksStore>().userUnblocked(person.id);
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(
+            content: Text('${person.preferredName} unblocked'),
+          ),
+        );
+      } on SocketException {
+        showSnackBar('Network error');
+        // ignore: avoid_catches_without_on_clauses
+      } on LemmyApiException catch (e) {
+        showSnackBar(e.message);
+      } catch (e) {
+        showSnackBar(e.toString());
+      } finally {
+        pending.value = false;
+      }
+    }
```

Comment:
This is exactly what we wanted to avoid by using MobX. If this widget requires such logic then it should have it's own store. Then you can use ready widgets such as `AsyncStoreListener` instead of duplicating handling logic.